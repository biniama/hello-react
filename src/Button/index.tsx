import React, { FC, memo, MouseEvent } from "react";

interface ButtonProps {
  handleClick: Function
}

const Button: FC<ButtonProps> = ({ handleClick }) => {

  function onClick(e: MouseEvent<HTMLButtonElement>): void {
    handleClick();
  }

  return <button onClick={onClick}>Click me!!!</button>;
};

export default memo(Button);
