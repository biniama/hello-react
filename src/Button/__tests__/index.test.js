import React from "react";
import snapshotRenderer from "react-test-renderer";
import Button from "..";

describe("Button component", () => {
  it("matches Button snapshot", () => {
    const component = snapshotRenderer.create(
      <Button handleClick={jest.fn()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
