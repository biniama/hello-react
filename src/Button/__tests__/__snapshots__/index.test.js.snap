// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches Button snapshot 1`] = `
<button
  onClick={[Function]}
>
  Click me!!!
</button>
`;
